File: xposedfirstmodule/src/main/java/me/gg/xposedfirstmodule/MainActivity.java
package me.gg.xposedfirstmodule;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private TextView tv_ad;
    private Button btn_load_ad;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tv_ad = findViewById(R.id.tv_ad);
        btn_load_ad = findViewById(R.id.btn_load_ad);

        btn_load_ad.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                tv_ad.setText(getTTAd());
            }
        });
    }


    public String getTTAd(){
        return "广告加载成功";
    }
}

File: app/src/main/java/me/gg/helloworld2018/MainActivity.kt
package me.gg.helloworld2018

import android.app.Activity
import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val bundle = this.intent.extras
        if (bundle != null && bundle.containsKey("email") ) {
            var str = bundle["email"].toString() + ":" + bundle["password"]
            Log.i("调试信息GREAT: ", str)
            this.txtMainInfo.text = str
//        Toast.makeText(this, str, Toast.LENGTH_LONG)
        }

//        this.findViewById<Button>(R.id.btnMainFinish)
        btnMainFinish.setOnClickListener {
            //            Toast.makeText(this, txtMainInfo.text, Toast.LENGTH_LONG)
            var intent = Intent()
            intent.putExtra("result", "这是返回结果内容:" + txtFoo.text)
            this.setResult(Activity.RESULT_OK, intent)
            this.finish()
        }

        this.btnShowFoo.setOnClickListener {
            Toast.makeText(this, txtFoo.text, Toast.LENGTH_LONG).show()
        }

    }

}
